In [7]:
cp Practicas/BaseDatos/TLALPAN/sectorH/names_2009-08-30.txt ./Tareas/Resultado

In [9]:
cat < names_2009-08-30.txt > names_cat.txt

In [10]:
cat names_cat.txt

Aguilar Natasha Alejanda	5908405556

Salomon Valeria	5550562274

Cedillo Arath Abijah	5957438808

Lazaro Heriberto	5131681455

Muñoz Maria Renee	5289217179

Saldaña Francisco	5393306394

Ramos Carmen Sofía	5564653521

Buneder Erick Ricardo	5853411637

Balandra Ana Ximena	5203278041

Molina Eduardo Alberto	5736392724

Barranco Carol Sthefany	5664011544

Santiago Lesli Paola	5291161226

Alba Anabel	5642845849

Aguilar Alexia	5462563799

Serralde Angel Sebastian	5522588882

Yines Ana Ximena	5593340690

Calette Alexia	5032726714

Balmaceda Jose De	5214401999

Cerezo Jesica Jocelyn	5003217550

Garnica Kevin	5978399615



Envía el archivo *names_2009-08-30.txt* como input al comando *cat*, y el resultado de eso (que es la información de names_2009-08-30.txt) lo envía como output al archivo *names_cat*. Como en este caso no existía simplemente lo creó.

Hubiera pasado lo mismo escribiendo solamente *cat names_2009-08-30.txt > names_cat*.

In [22]:
cat < names_2009-08-30.txt >> names_cat.txt

In [23]:
cat names_cat.txt

Aguilar Natasha Alejanda	5908405556

Salomon Valeria	5550562274

Cedillo Arath Abijah	5957438808

Lazaro Heriberto	5131681455

Muñoz Maria Renee	5289217179

Saldaña Francisco	5393306394

Ramos Carmen Sofía	5564653521

Buneder Erick Ricardo	5853411637

Balandra Ana Ximena	5203278041

Molina Eduardo Alberto	5736392724

Barranco Carol Sthefany	5664011544

Santiago Lesli Paola	5291161226

Alba Anabel	5642845849

Aguilar Alexia	5462563799

Serralde Angel Sebastian	5522588882

Yines Ana Ximena	5593340690

Calette Alexia	5032726714

Balmaceda Jose De	5214401999

Cerezo Jesica Jocelyn	5003217550

Garnica Kevin	5978399615

Aguilar Natasha Alejanda	5908405556

Salomon Valeria	5550562274

Cedillo Arath Abijah	5957438808

Lazaro Heriberto	5131681455

Muñoz Maria Renee	5289217179

Saldaña Francisco	5393306394

Ramos Carmen Sofía	5564653521

Buneder Erick Ricardo	5853411637

Balandra Ana Ximena	5203278041

Molina Eduardo Alberto	5736392724

Barranco Carol Sthefany	5664011544

Santiago Lesli Paola	52

Hace lo mismo que el anterior, solo que *>>* es para concatenar la información de *names_2009-08-30* al final de *names_cat.txt*, sin sobreescribir lo que ya existía. En este caso simplemente duplicamos la información.

La diferencia entre > y >> es que el primero sobreescribe la información que ya existía y el segundo no.

In [ ]:
cat names.txt 2> err.dat

In [ ]:
cata names.txt 2> err2.dat

La opción **2>** se usa para redirigir stderr (standard error), donde bash almacena los errores. Los comandos anteriores son inválidos pero tienen diferentes errores, así que el contenido de de *err.dat* y *err2.dat* es distinto.

In [1]:
cat err.dat

cat: names.txt: No such file or directory


In [41]:
cat err2.dat


Command 'cata' not found, did you mean:

  command 'caja' from deb caja (1.24.0-1)
  command 'cat' from deb coreutils (8.30-3ubuntu2)

Try: sudo apt install <deb name>



Ejecutar *cat > feliz* (input) ctrl+D ctrl+D Copia lo que se escriba como input al archivo *feliz*.

In [42]:
cat feliz

The cake is a lie. 



Para ver archivos en orden inverso podemos usar el comando **tac.** (**cat** escrito al revés).

In [35]:
cat BaseDatos/COYOACAN/sectorG/names_2005-09-27.dat

Marino Jorge Gabriel	5083218841

Calle Jose Manuel	5267956463

Gutierrez Maria Renee	5850364252

Aranda Melissa	5178803725

Adame Mario Arturo	5600141012

Icelo German	5166787251

Estrada Arturo	5636815256

Cruz Arath Abijah	5816462199

Alexeev Alexia	5715139951

Piccini Andrea	5829648811

Perea Maria Jose	5841034141

Goitia Carolina	5717864060

Rivera Yanko Salvador	5474309681

Iturriaga Adan Israel	5210580667

Calcaneo Abiram	5781375705

Galindo Rodrigo	5292735418

Arredondo Jorge Alberto	5944048778

Catalan Martha Mariana	5274177518

Penilla Emmanuel	5990448441

Lopez Martin	5270565437



In [9]:
tac Practicas/BaseDatos/COYOACAN/sectorG/names_2005-09-27.dat > catinverso.dat

In [10]:
cat catinverso.dat


Lopez Martin	5270565437

Penilla Emmanuel	5990448441

Catalan Martha Mariana	5274177518

Arredondo Jorge Alberto	5944048778

Galindo Rodrigo	5292735418

Calcaneo Abiram	5781375705

Iturriaga Adan Israel	5210580667

Rivera Yanko Salvador	5474309681

Goitia Carolina	5717864060

Perea Maria Jose	5841034141

Piccini Andrea	5829648811

Alexeev Alexia	5715139951

Cruz Arath Abijah	5816462199

Estrada Arturo	5636815256

Icelo German	5166787251

Adame Mario Arturo	5600141012

Aranda Melissa	5178803725

Gutierrez Maria Renee	5850364252

Calle Jose Manuel	5267956463

Marino Jorge Gabriel	5083218841


In [13]:
cat ../../Practicas/BaseDatos/CUAUHTEMOC/sectorE/* > todosarchivos.txt

In [14]:
cat todosarchivos.txt

Razo Pedro	5078328708

Zavala Oscar Enrique	5349803565

Arevalo Anabel	5098283547

Castillo Regina	5585642291

Muñoz Alexia	5670317235

Molina Montserrat	5506646832

Paredes Jose Manuel	5683845245

Meza Jovanna Sofia	5197111714

Hernandez Ana Karen	5000717497

Cejudo Elsy Daniela	5571564174

Sarabia Ana Ximena	5563916832

Cerezo Santiago Miguel	5018574468

Mendez Abiram	5908264097

Vidaños Jose De	5595988687

Cedillo Jesica Jocelyn	5948863480

Romero Alexis	5584398354

Sierra Heriberto	5497822462

Contreras Logan Alexander	5021410239

Munguia Luis Fernando	5120254168

Orozco María Fernanda	5469440435

Rosales Maroon Liliana	5459481540

Yines Aimee Berenice	5594659911

Marquez Kevin Kaled	5322379413

Caballero Jose De	5866484113

Solís Jesus	5954572200

Lagunas Gerardo	5543595417

Fernandez Will Cameron	5410185674

Saldaña Mariana Camila	5767575914

Mendez Emmanuel	5263539085

Corona Ana Paola	5225467712

Piña Francisco Alexander	5028397432

Silva Aimee Berenice	5381719174

Manriquez Ma

In [3]:
head -23 todosarchivos.txt | cat -n > colacabeza

In [3]:
tail -32 todosarchivos.txt | cat -n >> colacabeza

In [9]:
cat colacabeza

     1	Razo Pedro	5078328708
     2	
     3	Zavala Oscar Enrique	5349803565
     4	
     5	Arevalo Anabel	5098283547
     6	
     7	Castillo Regina	5585642291
     8	
     9	Muñoz Alexia	5670317235
    10	
    11	Molina Montserrat	5506646832
    12	
    13	Paredes Jose Manuel	5683845245
    14	
    15	Meza Jovanna Sofia	5197111714
    16	
    17	Hernandez Ana Karen	5000717497
    18	
    19	Cejudo Elsy Daniela	5571564174
    20	
    21	Sarabia Ana Ximena	5563916832
    22	
    23	Cerezo Santiago Miguel	5018574468
     1	Maya Ximena Itzel	5187140592
     2	
     3	Cazares Heriberto	5591260956
     4	
     5	Saldaña Katia Mariana	5297695281
     6	
     7	Balandra Mabel	5346871896
     8	
     9	Aguilar Maria Fernanda	5620980938
    10	
    11	Cazares Ximena	5031790603
    12	
    13	Nieto Oscar Enrique	5529321728
    14	
    15	Ruiz Alexis	5330479468
    16	
    17	Serrano Luis Fernando	5429722585
    18	
    19	Avila Monserrat	5896435866
    20	
    21	Ramos Martha	5849299706
    22	
 

El comando **umask** se usa para determinar la configuración default de permisos al crear archivos nuevos. Puede especificarse tanto de forma simbólica como numérica. Si se especifican en formato octal debe ponerse un *0* al inicio para especificar que los números siguientes se refieren a permisos.

Por ejemplo:

In [ ]:
umask 0664

Especifica que el dueño y el grupo tendrán permisos de lectura y escritura, y el resto sólo de lectura por default para cualquier archivo nuevo.

In [4]:
cp -r ../../Practicas/BaseDatos/CUAUHTEMOC .

In [25]:
chmod 614 CUAUHTEMOC/*/*

In [26]:
ls -Rl CUAUHTEMOC/

CUAUHTEMOC/:
total 44
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorA
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorE
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorG
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorJ
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorM
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorN
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorO
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorP
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorR
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorU
drwx--xr-- 2 zarathustra zarathustra 4096 nov  5 15:33 sectorV

CUAUHTEMOC/sectorA:
total 80
-rw---xr-- 1 zarathustra zarathustra 618 nov  5 15:33 names_2000-07-13.txt
-rw---xr-- 1 zarathustra zarathustra 627 nov  5 15:33 names_2001-04-24.txt
-rw---xr-- 1 zarathustra zarathustra 634 nov  5 15:33 names_2002-12-26.dat
-rw---xr-- 1 zarathustra zarat

In [8]:
chmod 740 BaseDatos/CUAUHTEMOC/sectorG/*.txt

In [9]:
ls -la BaseDatos/CUAUHTEMOC/sectorG/

total 168
drwxrwxr-x  2 zarathustra zarathustra 4096 oct 29 13:49 .
drwxrwxr-x 13 zarathustra zarathustra 4096 oct 29 13:49 ..
-rwxr-----  1 zarathustra zarathustra  623 oct 29 13:49 names_2000-04-14.txt
-rw-rw-r--  1 zarathustra zarathustra  638 oct 29 13:49 names_2001-03-03.dat
-rwxr-----  1 zarathustra zarathustra  626 oct 29 13:49 names_2001-10-21.txt
-rwxr-----  1 zarathustra zarathustra  626 oct 29 13:49 names_2002-01-07.txt
-rwxr-----  1 zarathustra zarathustra  658 oct 29 13:49 names_2002-06-03.txt
-rwxr-----  1 zarathustra zarathustra  611 oct 29 13:49 names_2002-07-31.txt
-rwxr-----  1 zarathustra zarathustra  608 oct 29 13:49 names_2002-12-03.txt
-rwxr-----  1 zarathustra zarathustra  641 oct 29 13:49 names_2003-06-06.txt
-rw-rw-r--  1 zarathustra zarathustra  643 oct 29 13:49 names_2003-09-09.dat
-rwxr-----  1 zarathustra zarathustra  609 oct 29 13:49 names_2005-02-27.txt
-rwxr-----  1 zarathustra zarathustra  628 oct 29 13:49 names_2005-06-11.txt
-rwxr-----  1 zarathustra 

In [10]:
chmod 714 BaseDatos/CUAUHTEMOC/sectorP/*.dat

In [11]:
ls -la BaseDatos/CUAUHTEMOC/sectorP/

total 168
drwxrwxr-x  2 zarathustra zarathustra 4096 oct 29 13:49 .
drwxrwxr-x 13 zarathustra zarathustra 4096 oct 29 13:49 ..
-rwx--xr--  1 zarathustra zarathustra  643 oct 29 13:49 names_2000-03-28.dat
-rwx--xr--  1 zarathustra zarathustra  613 oct 29 13:49 names_2000-06-06.dat
-rwx--xr--  1 zarathustra zarathustra  620 oct 29 13:49 names_2000-10-02.dat
-rwx--xr--  1 zarathustra zarathustra  596 oct 29 13:49 names_2000-11-24.dat
-rwx--xr--  1 zarathustra zarathustra  605 oct 29 13:49 names_2000-11-27.dat
-rwx--xr--  1 zarathustra zarathustra  660 oct 29 13:49 names_2001-02-27.dat
-rw-rw-r--  1 zarathustra zarathustra  619 oct 29 13:49 names_2001-12-09.txt
-rwx--xr--  1 zarathustra zarathustra  632 oct 29 13:49 names_2002-02-05.dat
-rwx--xr--  1 zarathustra zarathustra  612 oct 29 13:49 names_2002-08-11.dat
-rwx--xr--  1 zarathustra zarathustra  612 oct 29 13:49 names_2004-09-20.dat
-rwx--xr--  1 zarathustra zarathustra  638 oct 29 13:49 names_2006-09-19.dat
-rw-rw-r--  1 zarathustra 

In [14]:
ls -la ../../Practicas/BaseDatos/CUAUHTEMOC/sectorP/ > permisos.dat

In [16]:
chmod 700 BaseDatos/CUAUHTEMOC/sectorP/*-05-*

In [17]:
chmod 700 BaseDatos/CUAUHTEMOC/sectorO/*-05-*

In [18]:
ls -la BaseDatos/CUAUHTEMOC/sectorP

total 168
drwxrwxr-x  2 zarathustra zarathustra 4096 oct 29 13:49 .
drwxrwxr-x 13 zarathustra zarathustra 4096 oct 29 13:49 ..
-rwx--xr--  1 zarathustra zarathustra  643 oct 29 13:49 names_2000-03-28.dat
-rwx--xr--  1 zarathustra zarathustra  613 oct 29 13:49 names_2000-06-06.dat
-rwx--xr--  1 zarathustra zarathustra  620 oct 29 13:49 names_2000-10-02.dat
-rwx--xr--  1 zarathustra zarathustra  596 oct 29 13:49 names_2000-11-24.dat
-rwx--xr--  1 zarathustra zarathustra  605 oct 29 13:49 names_2000-11-27.dat
-rwx--xr--  1 zarathustra zarathustra  660 oct 29 13:49 names_2001-02-27.dat
-rw-rw-r--  1 zarathustra zarathustra  619 oct 29 13:49 names_2001-12-09.txt
-rwx--xr--  1 zarathustra zarathustra  632 oct 29 13:49 names_2002-02-05.dat
-rwx--xr--  1 zarathustra zarathustra  612 oct 29 13:49 names_2002-08-11.dat
-rwx--xr--  1 zarathustra zarathustra  612 oct 29 13:49 names_2004-09-20.dat
-rwx--xr--  1 zarathustra zarathustra  638 oct 29 13:49 names_2006-09-19.dat
-rw-rw-r--  1 zarathustra 

In [19]:
ls -la BaseDatos/CUAUHTEMOC/sectorO

total 88
drwxrwxr-x  2 zarathustra zarathustra 4096 oct 29 13:49 .
drwxrwxr-x 13 zarathustra zarathustra 4096 oct 29 13:49 ..
-rw-rw-r--  1 zarathustra zarathustra  652 oct 29 13:49 names_2000-08-09.dat
-rw-rw-r--  1 zarathustra zarathustra  617 oct 29 13:49 names_2002-01-15.dat
-rwx------  1 zarathustra zarathustra  618 oct 29 13:49 names_2003-05-04.dat
-rw-rw-r--  1 zarathustra zarathustra  615 oct 29 13:49 names_2004-03-16.txt
-rw-rw-r--  1 zarathustra zarathustra  635 oct 29 13:49 names_2004-04-19.dat
-rw-rw-r--  1 zarathustra zarathustra  629 oct 29 13:49 names_2004-10-29.dat
-rw-rw-r--  1 zarathustra zarathustra  633 oct 29 13:49 names_2005-10-07.dat
-rw-rw-r--  1 zarathustra zarathustra  657 oct 29 13:49 names_2006-02-09.txt
-rw-rw-r--  1 zarathustra zarathustra  626 oct 29 13:49 names_2006-02-22.txt
-rw-rw-r--  1 zarathustra zarathustra  628 oct 29 13:49 names_2009-01-12.txt
-rwx------  1 zarathustra zarathustra  627 oct 29 13:49 names_2009-05-06.dat
-rw-rw-r--  1 zarathustra z

In [7]:
find BaseDatos/ -name "*.txt" | wc -l

2036


Hay 2036 archivos txt en la base de datos.

In [8]:
find BaseDatos/ -name "*.dat" | wc -l

2121


Hay 2121 archivos .dat en la base de datos.

In [25]:
grep -rw "Daniel" BaseDatos/ | wc -l

573


Mi nombre aparece 573 veces en total. Uso la opción *-w* para excluir matches del tipo "Daniela", por ejemplo, y asegurar que sean solo las líneas con mi nombre exacto.

In [37]:
grep -rwL "Daniel" BaseDatos/ | wc -l

3615


Hay 3615 archivos en los que no aparece mi nombre. También intenté usar **grep -rwvl**, pero creo que eso le dice a bash que imprima el nombre de cualquier archivo que contenga *al menos una línea* SIN el patrón. Como eso ocurre en todos los archivos me arrojaba 4157, que es la cantidad total de archivos en la base de datos.

In [30]:
find BaseDatos/ -type f | wc -l

4157


In [23]:
grep -wl "Daniel" BaseDatos/*/*/*2002* | wc -l

29


Mi nombre aparece en 29 archivos con mi año de nacimiento.

In [29]:
find BaseDatos/ -type d | wc -l

187


La base de datos contiene 187 directorios y subdirectorios en total.

In [31]:
find BaseDatos/ -type f -size -1150c | wc -l

4157


Todos los archivos pesan menos de 1150 bytes.

In [34]:
find BaseDatos/*/sectorS/ -name "*2020*" | wc -l

8


Hay 8 archivos en el sectorS (de cualquier alcaldía) que incluyen "2020" en su nombre.